src/games/strategy/ui/ProgressDialog.java

```diff
@@ -0,0 +1,75 @@
+package games.strategy.ui;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Frame;
+import java.util.Objects;
+
+import javax.swing.JDialog;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JProgressBar;
+import javax.swing.border.EmptyBorder;
+import javax.swing.border.LineBorder;
+
+/**
+ * A modal dialog used to display indeterminate progress during an operation.
+ */
+public final class ProgressDialog extends JDialog {
+  private static final long serialVersionUID = -590470596784214914L;
+
+  private final String message;
+
+  /**
+   * Initializes a new instance of the {@code ProgressDialog} class.
+   *
+   * @param owner The {@code Frame} from which the dialog is displayed or {@code null} to use a shared, hidden frame as
+   *        the owner of the dialog.
+   * @param message The progress message; must not be {@code null}.
+   */
+  public ProgressDialog(final Frame owner, final String message) {
+    super(owner, true);
+
+    Objects.requireNonNull(message);
```

Comment:
Caution when there are Guava exact equivalents to what is available now in Java8. For example, `Sets.newHashSet()` in Guava is deprecated now (it was there back when JDK6 or 7 were out, before empty diamon notation, ie: `new HashSet<>()`)

In those cases we should be using the java version. In this case though `checkNotNull` has additional functionality.